ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/DownloadHeadersStep.java

```diff
@@ -85,23 +95,52 @@ private CompletableFuture<List<BlockHeader>> downloadHeaders(final SyncTargetRan
               protocolSchedule,
               protocolContext,
               ethContext,
+              peerTaskExecutor,
+              synchronizerConfiguration,
               range.getEnd(),
               range.getSegmentLengthExclusive(),
               validationPolicy,
               metricsSystem)
           .run();
     } else {
       LOG.debug("Downloading headers starting from {}", range.getStart().getNumber());
-      return GetHeadersFromPeerByHashTask.startingAtHash(
-              protocolSchedule,
-              ethContext,
-              range.getStart().getHash(),
-              range.getStart().getNumber(),
-              headerRequestSize,
-              metricsSystem)
-          .assignPeer(range.getSyncTarget())
-          .run()
-          .thenApply(PeerTaskResult::getResult);
+      if (synchronizerConfiguration.isPeerTaskSystemEnabled()) {
+        return ethContext
+            .getScheduler()
+            .scheduleServiceTask(
+                () -> {
+                  LOG.info("Getting headers using peer task system");
+                  GetHeadersFromPeerTask task =
+                      new GetHeadersFromPeerTask(
+                          range.getStart().getHash(),
+                          range.getStart().getNumber(),
+                          headerRequestSize,
+                          0,
+                          GetHeadersFromPeerTask.Direction.FORWARD,
+                          protocolSchedule);
+                  PeerTaskExecutorResult<List<BlockHeader>> taskResult =
+                      peerTaskExecutor.executeAgainstPeer(task, range.getSyncTarget());
```

Comment:
We should not execute against the sync target! I can see that the original implementation does that, but it is wrong!!!

Reply:
Might be best if you raise a ticket to correct this. These changes must be as close in behavior to the old code as possible